src/tests/modules/lock.rs

```diff
@@ -0,0 +1,20 @@
+use std::sync::{LockResult, RwLock, RwLockReadGuard, RwLockWriteGuard};
+
+#[derive(Default)]
+pub struct TestLock {
+    inner: RwLock<()>,
+}
+
+impl TestLock {
+    pub fn new() -> Self {
+        Default::default()
+    }
+
+    pub fn run_concurrently(&self) -> LockResult<RwLockReadGuard<'_, ()>> {
+        self.inner.read()
```

Comment:
We should `unwrap` the calls to `read` and `write` here so that the tests fail loudly if we can't acquire the lock

Reply:
makes sense. fixed